File: backend/slotify/spring/src/main/java/com/slotify/backend/spring/user/dtos/UserSummary.java
package com.slotify.backend.spring.user.dtos;

import lombok.*;

import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Setter
@ToString
@Builder
public class UserSummary {

    private UUID userId;
    private String username;
    private String s3ImageUrl;
    private String phoneNumber;
    private String emailAddress;
    private LocalDateTime createdAt;
}

File: frontend/app/src/main/java/com/raj/slotify/dtos/user/UserSummary.kt
package com.raj.slotify.dtos.user

import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime
import java.util.UUID

data class UserSummary(
    @SerializedName("userId")
    val userId: UUID,

    @SerializedName("username")
    val username: String,

    @SerializedName("s3ImageUrl")
    val s3ImageUrl: String?,

    @SerializedName("phoneNumber")
    val phoneNumber: String,

    @SerializedName("emailAddress")
    val emailAddress: String,

    @SerializedName("createdAt")
    val createdAt: LocalDateTime
)
